From f07cc57a24f33a669fca9f70c32d4a3ce01352e9 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Fri, 8 Mar 2024 14:50:54 +0000
Subject: [PATCH] ARM: dts: rp1: Add a gpio-ranges property

gpio-ranges declares a relationship between pinctrl and GPIO
controllers. It is required for "strict" mode to work.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/broadcom/rp1.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/broadcom/rp1.dtsi b/arch/arm/boot/dts/broadcom/rp1.dtsi
index 965b614a5f0a..fd9fb2dde0f7 100644
--- a/arch/arm/boot/dts/broadcom/rp1.dtsi
+++ b/arch/arm/boot/dts/broadcom/rp1.dtsi
@@ -465,6 +465,7 @@ rp1_gpio: gpio@d0000 {
 			#gpio-cells = <2>;
 			interrupt-controller;
 			#interrupt-cells = <2>;
+			gpio-ranges = <&rp1_gpio 0 0 54>;
 
 			rp1_uart0_14_15: rp1_uart0_14_15 {
 				pin_txd {
-- 
2.34.1

